Sources_v3/Controllers/ChannelListController+SwiftUI.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+@available(iOS 13, *)
+extension ChannelListControllerGeneric {
+    /// A wrapper object for `ChannelListController` type which makes it possible to use the controller comfortably in SwiftUI.
+    public class ObservableObject: SwiftUI.ObservableObject {
+        /// The underlying controller. You can still access it and call methods on it.
+        public let controller: ChannelListControllerGeneric
```

Comment:
We have a retain-cycle here, don't we?

Reply:
🤔 , we have, however, here it's correct. The observable object have to keep the controller alive. The problem is here https://github.com/GetStream/stream-chat-swift/pull/447/files/c3dc0f4af62feda2e3dedbfb4611fca00135ceb9#diff-0581a70d6a01bca9f6e926b164ff74bfR80 . There it should be `weak` but that would be strange. Uff. I need to think about this.

Reply:
fixed in 33cea36